### PR TITLE
SONARIAC-408 [TF] S6383 Use Provider Version Sensitive Analysis in RoleBasedAccessControlCheck

### PR DESCRIPTION
**Pull request by nils-werner-sonarsource**

### CODE REVIEW

File: iac-extensions/terraform/src/main/java/org/sonar/iac/terraform/checks/azure/RoleBasedAccessControlCheck.java [hunk 1]
```diff
@@ -19,70 +19,61 @@
...
-   * Report if 'role_based_access_control->enabled' is set to false
-   * or check 'role_based_access_control->azure_active_directory' block
-   */
-  private static void checkRoleBasedAccessControl(CheckContext ctx, BlockTree rbac) {
-    PropertyUtils.get(rbac, "enabled", AttributeTree.class)
-      .filter(attr -> TextUtils.isValueFalse(attr.value()))
-      .ifPresentOrElse(attr -> ctx.reportIssue(attr, DISABLED_MESSAGE),
-        () -> checkAzureActiveDirectory(ctx, rbac));
-  }
+      AttributeSymbol rbacEnabled = rbac.attribute("enabled").reportIf(isFalse(), DISABLED_MESSAGE);
```
Comment (pynicolas) [inline iac-extensions/terraform/src/main/java/org/sonar/iac/terraform/checks/azure/RoleBasedAccessControlCheck.java:55]: Should we now also check `azure_active_directory_role_based_access_control.azure_rbac_enabled`?

File: iac-extensions/terraform/src/main/java/org/sonar/iac/terraform/checks/azure/RoleBasedAccessControlCheck.java [hunk 2]
```diff
@@ -19,70 +19,61 @@
...
-  protected void registerResourceChecks() {
-    register(RoleBasedAccessControlCheck::checkKubernetesCluster, "azurerm_kubernetes_cluster");
-    register(RoleBasedAccessControlCheck::checkKeyVault, "azurerm_key_vault");
-  }
+  protected void registerResourceConsumer() {
+    // Report if 'role_based_access_control' block is missing or check its attributes.
+    // Report if 'role_based_access_control.enabled' is set to false
+    // or check 'role_based_access_control.azure_active_directory' block
+    // Report if 'role_based_access_control.azure_active_directory.managed' is set to true
+    // and 'role_based_access_control.azure_active_directory.managed' is missing or set to false.
```
Comment (pynicolas) [inline iac-extensions/terraform/src/main/java/org/sonar/iac/terraform/checks/azure/RoleBasedAccessControlCheck.java:46]:
```suggestion
    // and 'role_based_access_control.azure_active_directory.azure_rbac_enabled' is missing or set to false.
```
Not sure about that one

File: iac-extensions/terraform/src/test/resources/checks/Azure/RoleBasedAccessControlCheck/kubernetes_cluster_version3.tf [hunk 3]
```diff
@@ -31,5 +31,25 @@ resource "azurerm_kubernetes_cluster" "role_based_access_control_enabled" {
...
+
+resource "azurerm_kubernetes_cluster" "incomplete_azure_active_directory_role_based_access_control" {
+  azure_active_directory_role_based_access_control {
+    managed = true # Noncompliant {{Omitting 'azure_rbac_enabled' disables role-based access control for this resource. Make sure it is safe here.}}
+  }
+}
+
+resource "azurerm_kubernetes_cluster" "compliant_azure_active_directory_role_based_access_control" {
+  azure_active_directory_role_based_access_control {
+    managed = flase
```
Comment (pynicolas) [inline iac-extensions/terraform/src/test/resources/checks/Azure/RoleBasedAccessControlCheck/kubernetes_cluster_version3.tf:49]: Is the typo intended?

File: iac-extensions/terraform/src/test/resources/checks/Azure/RoleBasedAccessControlCheck/kubernetes_cluster_version3.tf [hunk 4]
```diff
@@ -31,5 +31,25 @@ resource "azurerm_kubernetes_cluster" "role_based_access_control_enabled" {
...
+resource "azurerm_kubernetes_cluster" "noncompliant_azure_active_directory_role_based_access_control" {
+  azure_active_directory_role_based_access_control {
+    managed = true
+    azure_rbac_enabled = false # Noncompliant {{Make sure that disabling role-based access control is safe here.}}
+  }
+}
+
+resource "azurerm_kubernetes_cluster" "incomplete_azure_active_directory_role_based_access_control" {
+  azure_active_directory_role_based_access_control {
+    managed = true # Noncompliant {{Omitting 'azure_rbac_enabled' disables role-based access control for this resource. Make sure it is safe here.}}
```
Comment (pynicolas) [inline iac-extensions/terraform/src/test/resources/checks/Azure/RoleBasedAccessControlCheck/kubernetes_cluster_version3.tf:43]: I guess that you managed to find what's the default value for `azure_active_directory_role_based_access_control.azure_rbac_enabled`, I don't see it on the [azurerm_kubernetes_cluster](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/kubernetes_cluster) documentation.

**Review by pynicolas — APPROVED** [review]

File: iac-extensions/terraform/src/test/resources/checks/Azure/RoleBasedAccessControlCheck/kubernetes_cluster_version3.tf (hunk 4 above)
Comment (nils-werner-sonarsource) [reply iac-extensions/terraform/src/test/resources/checks/Azure/RoleBasedAccessControlCheck/kubernetes_cluster_version3.tf:43]: I checked the [implementation](https://github.com/hashicorp/terraform-provider-azurerm/blob/29e5835f814827186d0e943562926357c93227a0/internal/services/containers/kubernetes_cluster_data_source.go#L1077) and IMO the default value seems to be false in any case.

**Pull request merged**